src/script.c

```diff
@@ -443,6 +443,10 @@ static int scriptVerifyClusterState(scriptRunCtx *run_ctx, client *c, client *or
             *err = sdscatfmt(sdsempty(),
                              "Command '%S' in script attempted to access keys that don't hash to the same slot",
                              c->cmd->fullname);
+        } else if (error_code == CLUSTER_REDIR_CROSS_SHARD) {
```

Comment:
I actually think the else case error is more useful, because we are trying to access a key from a non-local node.